dask/utils.py

```diff
@@ -806,11 +807,9 @@ def insert(tup, loc, val):
 
 
 def dependency_depth(dsk):
-    import toolz
-
     deps, _ = get_deps(dsk)
 
-    @toolz.memoize
+    @lru_cache()
```

Comment:
`toolz.memoize` doesn't have a limit on the size of the underlying cache, while `functools.lru_cache` uses of cache of 128 items by default. Could you specify `maxsize=None` here to keep the same behavior as `toolz.memoize`:

```suggestion
    @lru_cache(maxsize=None)
```